tests/Permissiion/PermissionTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+use App\Models\Permission;
+use App\Models\Role;
+use App\Models\User;
+use App\Repositories\RoleRepository;
+use App\Repositories\UserRepository;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Tests\TestCase;
+
+class PermissionTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+    }
+
+    /**
+     * @test
+     */
+    public function test_can_delete_role_with_manage_role_permission()
+    {
+        $this->makeUserWithPermissions(['manage_roles'], ['name' => 'librarian'])->id;
+
+        /** @var Role $role */
+        $role = factory(Role::class)->create();
+
+        $response = $this->deleteJson(route('roles.destroy', $role->id));
+
+        $response->assertStatus(200);
+    }
+
+    /**
+     * @test
+     */
+    public function test_unable_to_delete_role_without_manage_role_permission()
+    {
+        /** @var Role $role */
+        $role = factory(Role::class)->create();
+
+        $response = $this->deleteJson(route('roles.destroy', $role->id));
+
+        $this->assertExceptionMessage($response, 'Unauthenticated.');
+    }
+
+    /**
+     * @param array $permissions
+     * @param array $roleFields
+     *
+     * @return User
+     */
+    public function makeUserWithPermissions($permissions = [], $roleFields = [])
+    {
+        /** @var UserRepository $userRepo */
+        $userRepo = \App::make(UserRepository::class);
```

Comment:
in this function only following process are done.

1. create user
2. create a role
3. if permission exist then assign that permission to role
4. assign role to user that enough.

remove remaining code